### ISSUE TITLE
Crash with BaseMetals

### ISSUE THREAD

**Issue by D3nnis3n** [Lance5057/Tinkers-Defense#51]
"Platinum is already registered". As Base Metals and it's addons Register a hell lot of metals (around 100) this likeley happens to more metals when that one is fixed, you might want to look into that.

> net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Tinkers' Defense (tinkersdefense)
> Caused by: slimeknights.tconstruct.library.TinkerAPIException: [TCon API] Could not register Material "platinum": It was already registered by Base Metals
> 	at slimeknights.tconstruct.library.TinkerRegistry.error(TinkerRegistry.java:951)
> 	at slimeknights.tconstruct.library.TinkerRegistry.addMaterial(TinkerRegistry.java:129)
> 	at slimeknights.tconstruct.library.MaterialIntegration.preInit(MaterialIntegration.java:101)
> 	at lance5057.tDefense.core.materials.TDMaterials.createMaterial(TDMaterials.java:508)
> 	at lance5057.tDefense.core.materials.TDMaterials.preInit(TDMaterials.java:421)
> 	at lance5057.tDefense.TinkersDefense.preInit(TinkersDefense.java:287)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:606)
> 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
> 	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
> 	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
> 	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
> 	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
> 	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
> 	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:253)
> 	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:231)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
> 	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
> 	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
> 	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
> 	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
> 	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
> 	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
> 	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:148)
> 	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:602)
> 	at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:270)
> 	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:466)
> 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:377)
> 	at net.minecraft.client.main.Main.main(SourceFile:123)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.lang.reflect.Method.invoke(Unknown Source)
> 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
> 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
>

**Comment by Keetarin** [comment]
I agree with this guy, besides it would be a simple matter in my opinion (I don't code, so, correct me if i'm wrong). First detect if Base Metals and it's addons are installed, second if any of them are installed run an alternate code for the relevant metals.

**Comment by Chupp4-Jo3** [comment]
Just had this issue and it came up again blaming Moar Tinkers after disabling Tic Defence. I'm guessing it's the TicAPI itself and it's blaming the first addon mod that loads it but i have no idea.

https://pastebin.com/KJsaRNxL / https://pastebin.com/TgfYsqMH

**Comment by dshadowwolf** [comment]
(BaseMetals lead dev here) This issue actually arises when a mod doesn't check to see if a material of a given name is already registered when it tries to add its own material to Tinkers'

In the *Metals code we actually have a TinkersRegistry query to see if a material of a given name already exists before we try to register it. Possibly because of load-order issues this has had some issues - including conflicting with Tinkers' itself at one point (over Lead, of all things). However... Once we ironed out the issues that were letting BaseMetals register things with Tinkers' before Tinkers' was able to register with itself this has not been a major issue.

Until recently, with PlusTiC and Moar Tinkers' being ported to 1.12 and crashes happening with colliding trait names (resolved by changing the internal names of them in *Metals), colliding modifier names (same solution as trait names) and now colliding materials - With materials, as the name is used in a lot of places in *Metals beyond the Tinkers' code, we cannot take the route of changing the names (and it'd cause other issues if we did).

Basically... I'd like to suggest that all people integrating with Tinkers' start adding a TinkersRegistry query about materials before trying to register their own material.
